cmd/gateway-azure.go

```diff
@@ -304,12 +345,24 @@ func (a *azureObjects) ListObjects(bucket, prefix, marker, delimiter string, max
 			ContentEncoding: object.Properties.ContentEncoding,
 		})
 	}
+
+	// Remove minio.sys.tmp prefix.
+	for i, prefix := range resp.BlobPrefixes {
+		if prefix == globalMinioSysTmpPrefix {
+			resp.BlobPrefixes = append(resp.BlobPrefixes[:i], resp.BlobPrefixes[i+1:]...)
+			break
+		}
+	}
```

Comment:
We need not blindly remove these entries. because it will help us debugging if we allow listing when we do:
`mc ls myazure/testbucket/minio.sys.tmp/`

i.e we should not filter out the entries if the user is specifically listing under the minio.sys.tmp prefix.

You can take a look at gateway-gcs.go ListObjects on how we take care of this

Reply:
Done on appropriate section of the code.